### PR TITLE
createtestorder API had been added

### PR DESCRIPTION
**Pull request by ralsalloum**
- createtestorder API had been added.

### CODE REVIEW

File: backend/src/Constant/Order/OrderCostDefaultValueConstant.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace App\Constant\Order;
+
+final class OrderCostDefaultValueConstant
+{
+    const ORDER_VALUE_CONST = 15;
```
Comment (Kenan-Hussein) [inline backend/src/Constant/Order/OrderCostDefaultValueConstant.php]: This should be array also. like the location

File: backend/src/Constant/Order/OrderNoteDefaultValueConstant.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace App\Constant\Order;
+
+final class OrderNoteDefaultValueConstant
+{
+    const ORDER_NOTE_DEFAULT_VALUE_CONST = "Order made for testing";
```
Comment (Kenan-Hussein) [inline backend/src/Constant/Order/OrderNoteDefaultValueConstant.php]: it's cool to make this also an array, add some fancy crazy legendary notes

File: backend/src/Constant/Order/OrderRecipientDefaultValueConstant.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Constant\Order;
+
+final class OrderRecipientDefaultValueConstant
+{
+    const ORDER_RECIPIENT_NAME_DEFAULT_VALUE_CONST = "rami";
```
Comment (Kenan-Hussein) [inline backend/src/Constant/Order/OrderRecipientDefaultValueConstant.php]: same the above, show us what names you love most <3

File: backend/src/Request/Admin/Order/OrderCreateByAdminRequest.php [hunk 4]
```diff
@@ -133,4 +133,74 @@ public function setIsHide($isHide)
 
         return $this;
     }
+
```
Comment (Kenan-Hussein) [inline backend/src/Request/Admin/Order/OrderCreateByAdminRequest.php:136]: You should use brand new request object please don't use anything already exist

File: backend/src/Service/Admin/Order/AdminOrderService.php [hunk 5]
```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
...
+
+                // 10 create local notification for the store
+                $this->createLocalNotificationForStore($storeOwnerProfile->getId(), NotificationConstant::NEW_ORDER_TITLE,
+                    NotificationConstant::CREATE_ORDER_SUCCESS, $order->getId());
+
+                // 11 create order timeline
+                $this->createOrderLogViaOrderEntity($order);
+
+                // 12 save log of the action on order
+                $this->createOrderLogMessageViaOrderEntityAndByAdmin($order, $userId);
```
Comment (Kenan-Hussein) [inline backend/src/Service/Admin/Order/AdminOrderService.php]: What is this?

File: backend/src/Service/Admin/Order/AdminOrderService.php [hunk 6]
```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
...
+
+                // 13 prepare the response
+                $response[] = $this->autoMapping->map(OrderEntity::class, TestingOrderCreateByAdminResponse::class, $order);
+
+                // 14 create firebase notification to store
+                $this->sendFirebaseNotificationAboutOrderStateForUserByAdmin($storeOwnerProfile->getStoreOwnerId(),
```
Comment (Kenan-Hussein) [inline backend/src/Service/Admin/Order/AdminOrderService.php]: No need

File: backend/src/Service/Admin/Order/AdminOrderService.php [hunk 7]
```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
...
+                    $order->getId(), $order->getState(), NotificationConstant::STORE);
+
+                // 15 create firebase notification to captains
+                $this->sendFirebaseNotificationToAllCaptainsByAdmin($order->getId());
```
Comment (Kenan-Hussein) [inline backend/src/Service/Admin/Order/AdminOrderService.php]: no need those firebase calls will make the response needs 3 months to return

File: backend/src/Service/Admin/Order/AdminOrderService.php [hunk 8]
```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
...
+        $branch = $this->storeOwnerBranchService->getBranchEntityById($request->getBranch());
+
+        if (! $branch) {
+            return StoreOwnerBranch::BRANCH_NOT_FOUND;
+        }
+
+        $request->setStoreOwner($storeOwnerProfile);
+        $request->setBranch($branch);
+
+        for ($counter = 0; $counter < $request->getOrdersCount(); $counter++) {
```
Comment (Kenan-Hussein) [inline backend/src/Service/Admin/Order/AdminOrderService.php]: please check if there is '$request->getOrdersCount()" as a valid number >=1

File: backend/src/Service/Admin/Order/AdminOrderService.php [hunk 9]
```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
...
+                $this->sendFirebaseNotificationAboutOrderStateForUserByAdmin($storeOwnerProfile->getStoreOwnerId(),
+                    $order->getId(), $order->getState(), NotificationConstant::STORE);
+
+                // 15 create firebase notification to captains
+                $this->sendFirebaseNotificationToAllCaptainsByAdmin($order->getId());
+            }
+        }
+
+        return $response;
+    }
```
Comment (Kenan-Hussein) [inline backend/src/Service/Admin/Order/AdminOrderService.php]: This is very sad, please move everything related to this craziness to a new service.

File: backend/src/Service/Admin/Order/AdminOrderService.php [hunk 10]
```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
...
+            }
+
+            // 5 set the delivery date equals to current datetime + 3 minutes just for affording the late in persisting
+            // the order to the database if it is happened
+            $orderCreateByAdminRequest->setDeliveryDate((new DateTime('+ 3 minutes'))->format('Y-m-d H:i:s'));
+
+            $orderCreateByAdminRequest->setPayment(OrderTypeConstant::ORDER_PAYMENT_CASH);
+            $orderCreateByAdminRequest->setOrderCost(OrderCostDefaultValueConstant::ORDER_VALUE_CONST);
+            $orderCreateByAdminRequest->setNote(OrderNoteDefaultValueConstant::ORDER_NOTE_DEFAULT_VALUE_CONST);
+            $orderCreateByAdminRequest->setDestination(OrderClientLocationConstant::ORDER_CLIENT_LOCATION_ARRAY_CONST[array_rand(OrderClientLocationConstant::ORDER_CLIENT_LOCATION_ARRAY_CONST)]);
```
Comment (Kenan-Hussein) [inline backend/src/Service/Admin/Order/AdminOrderService.php]: Please move this 'random' to a separate function, as same as what you will do for the other random values.
This is for future upgrades to this sweet function.

File: backend/src/Service/Admin/Order/AdminOrderService.php [hunk 11]
```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
...
+            $orderCreateByAdminRequest->setRecipientName(OrderRecipientDefaultValueConstant::ORDER_RECIPIENT_NAME_DEFAULT_VALUE_CONST);
+            $orderCreateByAdminRequest->setRecipientPhone(OrderRecipientDefaultValueConstant::ORDER_RECIPIENT_PHONE_DEFAULT_VALUE_CONST);
+            $orderCreateByAdminRequest->setDetail(OrderNoteDefaultValueConstant::ORDER_NOTE_DEFAULT_VALUE_CONST);
+
+            // 6 calculate the distance between the store's branch and the client location
+            $distance = $this->getDistanceBetweenTwoLocations($branch->getLocation(), $orderCreateByAdminRequest->getDestination());
```
Comment (Kenan-Hussein) [inline backend/src/Service/Admin/Order/AdminOrderService.php]: does this call google api ?

File: backend/src/Constant/Order/OrderClientLocationConstant.php [hunk 12]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Constant\Order;
+
+final class OrderClientLocationConstant
+{
+    const ORDER_CLIENT_LOCATION_ARRAY_CONST = [
```
Comment (Kenan-Hussein) [inline backend/src/Constant/Order/OrderClientLocationConstant.php:7]: Not bad locations, but it better if we make them more distance

**Review by Kenan-Hussein — CHANGES_REQUESTED** [review]
Please resolve those easy-peasy comments

File: backend/src/Service/Admin/Order/AdminOrderService.php (hunk 5 above)
Comment (ralsalloum) [reply backend/src/Service/Admin/Order/AdminOrderService.php]: For creating order log via Symfony Messenger.

File: backend/src/Service/Admin/Order/AdminOrderService.php (hunk 8 above)
Comment (ralsalloum) [reply backend/src/Service/Admin/Order/AdminOrderService.php]: I have already checked it at the first _if_ block of the function.

File: backend/src/Service/Admin/Order/AdminOrderService.php (hunk 11 above)
Comment (ralsalloum) [reply backend/src/Service/Admin/Order/AdminOrderService.php]: Yub

**Review by Kenan-Hussein — APPROVED** [review]

**Pull request merged**